composer/utils/object_store/oci_object_store.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""S3-Compatible object store."""
+
+from __future__ import annotations
+
+import os
+import pathlib
+import uuid
+from typing import Callable, Optional, Union
+
+from composer.utils.import_helpers import MissingConditionalImportError
+from composer.utils.object_store.object_store import ObjectStore
+
+__all__ = ['OCIObjectStore']
+
+
+class OCIObjectStore(ObjectStore):
+    """Utility for uploading to and downloading from an OCI bucket.
+
+    Args:
+        bucket (str): The bucket name.
+        prefix (str): A path prefix such as `folder/subfolder/` to prepend to object names. Defaults to ''.
+    """
+
+    def __init__(
+        self,
+        bucket: str,
+        prefix: str = '',
+    ) -> None:
+        try:
+            import oci
+        except ImportError as e:
+            raise MissingConditionalImportError(conda_package='oci',
+                                                extra_deps_group='oci',
+                                                conda_channel='conda-forge') from e
+
+        # Format paths
+        self.bucket = bucket.strip('/')
+        self.prefix = prefix.strip('/')
+        if self.prefix:
+            self.prefix += '/'
+
+        config = oci.config.from_file()
+        self.client = oci.object_storage.ObjectStorageClient(config=config,
+                                                             retry_strategy=oci.retry.DEFAULT_RETRY_STRATEGY)
+        self.namespace = self.client.get_namespace().data
+
+    def get_uri(self, object_name: str) -> str:
+        return f'oci://{self.bucket}/{object_name}'
+
+    def get_object_size(self, object_name: str) -> int:
+        objects = self.client.list_objects(self.namespace, self.bucket, prefix=object_name, fields='size').data.objects
+        relevant_objects = [obj for obj in objects if obj.name == object_name]
+        if relevant_objects:
+            obj = relevant_objects.pop()
+            return obj.size
+        else:
+            raise FileNotFoundError(f'Unable to locate oci://{self.bucket}@{self.namespace}/{object_name}')
+
+    def upload_object(
+        self,
+        object_name: str,
+        filename: Union[str, pathlib.Path],
+        callback: Optional[Callable[[int, int], None]] = None,
+    ):
+        del callback
+        with open(filename, 'rb') as f:
```

Comment:
You can also use `UploadManager` rather than `put_object` which I guess allows parallel uploads as a chunk. Something like this

```
from oci.object_storage import UploadManager
upload_manager = UploadManager(object_storage_client, allow_parallel_uploads=True, parallel_process_count=3)
response = upload_manager.upload_file(namespace, bucket_name, object_name, filename)
```

Reply:
good call. done